### PR TITLE
Sof 1575/create rundown view header component

### PR DESCRIPTION
**Pull request by RasmusAlbrektsen**
This PR introduces the Rundown View Header component, as well as a context menu that can be accessed by right clicking the header. 

image

### CODE REVIEW

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="c-context-menu" cdkMenu>
+    <div class="c-context-menu-header">Options</div>
+    <button *ngIf="rundown?.isActive" class="c-context-menu-item" cdkMenuItem (click)="deactivate()">Deactivate</button>
+    <button *ngIf="rundown?.isActive" class="c-context-menu-item" cdkMenuItem (click)="take()">Take</button>
+    <button *ngIf="!rundown?.isActive"class="c-context-menu-item" cdkMenuItem (click)="activate()">Activate</button>
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html:5]: You could use templates for composing this.

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  flex-direction: row;
+  min-width: 64px;
+  align-items: center;
+  border: none;
+  line-height: 36px;
+  padding: 0 16px;
+
+  font-size: 1rem;
+
+  user-select: none;
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss:28]: `user-select` can be moved to a `:host` selector.

File: src/app/rundown/components/rundown-header/rundown-header.component.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<div [cdkContextMenuTriggerFor]="context_menu" class="c-header" [class.c-header--active]="rundown.isActive">
+    <ng-template #context_menu>
+    <sofie-rundown-header-context-menu class="c-context-access-surface" [rundown]="rundown"> </sofie-rundown-header-context-menu>
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.html:3]: Should be indented one level.

File: src/app/rundown/components/rundown-header/rundown-header.component.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    </ng-template>
+
+    <div class="c-information-row">
+        <div class="c-information-row-left">
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.html:7]: When naming css classes, we should aim to make descriptive names that describe their purpose, not how they are visually displayed.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+@Component({
+  selector: 'sofie-rundown-header',
+  templateUrl: './rundown-header.component.html',
+  styleUrls: ['./rundown-header.component.scss']
+})
+export class RundownHeaderComponent implements OnInit, OnDestroy, OnChanges {
+  @Input()
+  public rundown: Rundown
+
+  public showStyleVariant: ShowStyleVariant
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:17]: Do we always have a value for this attribute?

File: src/app/shared/components/header/header.component.scss [hunk 6]
```diff
@@ -2,6 +2,7 @@
   background-color: #343434;
   display: flex;
   flex-direction: row;
+  pointer-events: none;
```
Comment (KvelaGorrrrnio) [inline src/app/shared/components/header/header.component.scss:5]: Why is this added?

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        })
+        .then(unsubscribeFromShowStyleVariant => {
+          this.showStyleVariantSubscription = unsubscribeFromShowStyleVariant
+        })
+  }
+
+  public ngOnDestroy() {
+    clearInterval(this.updateCurrentLocalDateIntervalId)
+    this.showStyleVariantSubscription?.unsubscribe()
+  }
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:46]: newline would be nice :)

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  public setupName: string = ''
+  public schema: string = ''
+  public design: string = ''
+
+  constructor(private showStyleVariantStateService: ShowStyleVariantStateService) {}
+
+  public ngOnInit(): void {
+    this.updateCurrentLocalDateIntervalId = setInterval(() => {
+      this.currentLocalDate = new Date()
+    }, 1000)
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:31]: Curly brackets are unnecessary.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  public getShortenedRundownName(): string {
+    const rundownPathStrings: string[] = this.rundown?.name.split('.') ?? []
+    return rundownPathStrings[rundownPathStrings.length - 1]
+  }
+
+  private setDefaultHeaderInformation(): void {
+    const gfxDefaults = this.showStyleVariant.blueprintConfiguration.GfxDefaults[0]
+    this.setupName = gfxDefaults.DefaultSetupName.label
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:62]: What is this?

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    }, 1000)
+
+    this.showStyleVariantStateService
+        .subscribeToShowStyleVariant(this.rundown.id, (showStyleVariant) => {
+          this.showStyleVariant = showStyleVariant
+          this.setDefaultHeaderInformation()
+        })
+        .then(unsubscribeFromShowStyleVariant => {
+          this.showStyleVariantSubscription = unsubscribeFromShowStyleVariant
+        })
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:40]: Curly brackets are unnecessary.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+  private setSchemaFromInfinitePieces(): void {
+    const infinitePieces: Piece[] = this.rundown.infinitePieces
+
+    const schemaPiece: Piece | undefined = infinitePieces.find((piece) => this.getNameFromTemplate(piece.name) === 'SKEMA')
+    if(schemaPiece) {
+      this.schema = schemaPiece.name
+    }
+  }
+
+  private getNameFromTemplate(template: string) {
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:85]: This name seems a bit generic. Would it make sense to change it to something like `getGfxNameFromTemplate`?

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html (hunk 1 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html]: Templates have been put to use instead!

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss (hunk 2 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss]: It has been moved.

File: src/app/rundown/components/rundown-header/rundown-header.component.html (hunk 3 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.html]: Has been indented!

File: src/app/rundown/components/rundown-header/rundown-header.component.html (hunk 4 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.html]: This entire structure has now been changed.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 5 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:27]: Not if we don't receive a valid response from the backend. For that reason i'm changing it to optional and guarding it when we set the default values. Good catch.

File: src/app/shared/components/header/header.component.scss (hunk 6 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/components/header/header.component.scss]: Just a mistake i believe, it has been removed.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 7 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:69]: Inserted one just for you.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 8 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: Removed!

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 9 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:88]: It sets the default information received from the Show Style Variant. The GFX Setup, Design and Schema.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 10 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: Removed!

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 11 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: I have changed it to the suggestion!

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html [hunk 12]
```diff
@@ -0,0 +1,18 @@
+<div class="c-context-menu" cdkMenu>
+    <div class="c-context-menu-header">Options</div>
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html:2]: Shouldn't this be the name of the rundown??

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss [hunk 13]
```diff
@@ -0,0 +1,38 @@
+.c-context-menu {
+  display: inline-flex;
+  flex-direction: column;
+  min-width: 180px;
+  max-width: 280px;
+  background-color: rgb(255, 255, 255);
+  user-select: none;
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss:7]: For it to match the old context menu, it should have a slight border-radius.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 9 above)
Comment (KvelaGorrrrnio) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: Shouldn't this be showStyleVariant.name?

**Review by KvelaGorrrrnio — APPROVED** [review]

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html (hunk 12 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.html]: I have removed it.

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss (hunk 13 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.scss:9]: Added!

File: src/app/rundown/components/rundown/rundown.component.html [hunk 14]
```diff
@@ -1,30 +1,24 @@
-<body>
-<div class="title">{{rundown?.name}}</div>
+<sofie-rundown-header *ngIf="rundown" class="c-rundown-header" [rundown]="rundown"></sofie-rundown-header>
+
+<div *ngIf="rundown?.isActive" class="test-buttons">
+    <ng-container *ngIf="!rundown?.isActive; else activeRundownButtons"></ng-container>
+</div>
```
Comment (LindvedKrvang) [inline src/app/rundown/components/rundown/rundown.component.html:5]: If you negate the if-statement on line 3 then you can complete remove the if-statement on line 4 and write:
```
<div *ngIf=!rundown?.isActive ...>
  <button ...>Take Next</button>
</div>
```

**Review by LindvedKrvang — COMMENTED** [review]
It looks good :)

File: src/app/core/models/show-style-variant.ts [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+        },
+        DefaultSchema: {
+            value: string
+            label: string
+        },
+        DefaultDesign: {
+            value: string
+            label: string,
+        }
+    }[]
```
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/app/core/models/show-style-variant.ts:20]: Bonus info: The reason it's saved as an array is because of the "framework" Core provides to make the Settings page for Blurprints which forced us to use a "Table", which makes the entries an array.

File: src/app/core/services/show-style-variant-state.service.ts [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+        return [
+            this.connectionStatusObserver.subscribeToReconnect(this.resetShowStyleVariantSubjects.bind(this))
+        ]
+    }
+
+    private subscribeToShowStyleVariantEvents(): EventSubscription[] {
+        return []
+    }
+
+    private resetShowStyleVariantSubjects(): void {}
```
Comment (andr9528) [inline src/app/core/services/show-style-variant-state.service.ts:39]: Is this suppose to be an empty private method?

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts [hunk 17]
```diff
@@ -0,0 +1,37 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { RundownHeaderContextMenuComponent } from './rundown-header-context-menu.component';
+import {Rundown} from "../../../core/models/rundown";
+import {instance, mock, when} from "@typestrong/ts-mockito";
+import {RouterModule} from "@angular/router";
+import {RundownService} from "../../../core/abstractions/rundown.service";
+
+describe('HeaderContextMenuComponent', () => {
```
Comment (andr9528) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts:8]: Mismatch between tested component and text.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+    return piece.name.startsWith('DESIGN_')
+  }
+
+  private isSkemaInfinitePiece(piece: Piece): boolean {
+    return piece.name.startsWith('DESIGN_')
+  }
+
+  private getGfxNameFromTemplate(template: string) {
+    const pattern: RegExp = /^.+_(?<gfxName>\w+)$/
+    return template.match(pattern)?.groups?.['gfxName'] ?? template
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:114]: I don't see the benefit of extracting this to a constant.
If you don't want there to be a string literal, then I'll suggest doing something like:
```suggestion
    const { gfxName } = template.match(pattern)?.groups ?? {}
    return gfxName ?? template
```

File: src/app/rundown/components/rundown-header/rundown-header.component.html [hunk 19]
```diff
@@ -0,0 +1,23 @@
+<div [cdkContextMenuTriggerFor]="contextMenu" class="c-rundown-header" [class.c-rundown-header--active]="rundown.isActive">
+    <ng-template #contextMenu>
+        <sofie-rundown-header-context-menu class="c-context-access-surface" [rundown]="rundown"> </sofie-rundown-header-context-menu>
+    </ng-template>
+
+    <sofie-logo></sofie-logo>
+    <sofie-rundown-header-panel [label]="rundownPath">{{rundownName}}</sofie-rundown-header-panel>
```
Comment (andr9528) [inline src/app/rundown/components/rundown-header/rundown-header.component.html:7]: The label for this one is using camelCase, while the other labels use normal text.
Is there any reason for this?

Comment (andr9528) [reply src/app/rundown/components/rundown-header/rundown-header.component.html:7]: Is it because it uses the attribute in the component class?

Comment (andr9528) [reply src/app/rundown/components/rundown-header/rundown-header.component.html:7]: Looking at it further i am sure it is the reason.

But that now means that this is a magic string, unless i am mistaken.
Is this how it has been done in other places too?
I am not a fan of magic strings...

File: src/app/core/parsers/zod-entity-parser.service.ts [hunk 20]
```diff
@@ -6,8 +6,38 @@ import { Rundown } from '../models/rundown'
...
+            DefaultSchema: zod.object({
+                value: zod.string(),
+                label: zod.string(),
+            }),
+            DefaultDesign: zod.object({
+                value: zod.string(),
+                label: zod.string(),
+            }),
+        }).array()
+    })
```
[the remark this answers is not in the document]
Comment (andr9528) [reply src/app/core/parsers/zod-entity-parser.service.ts:32]: Perhaps it should just be explained in the README.md, such than any other company that wishes to fork it, knows that they will need to adjust certain parts to their Blueprints?

File: src/app/core/models/show-style-variant.ts [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+}
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/core/models/show-style-variant.ts:21]: I'm not sure how we accomplish that with they way we validate data going in to the frontend. I wasn't able to find a good way of doing it for this scenario.

File: src/app/core/parsers/zod-entity-parser.service.ts (hunk 20 above)
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/core/parsers/zod-entity-parser.service.ts:32]: It's a good point that we should discuss how we handle this, whether it be a more specific layer put on top so it can easily be recognized as specific to our own implementation or at least described somewhere.

File: src/app/core/services/show-style-variant-state.service.ts [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    private subscribeToEvents(): void {
+        this.eventSubscriptions = [
+            ...this.subscribeToShowStyleVariantEvents(),
+            ...this.subscribeToConnectionStatus(),
+        ]
+    }
+
+    private subscribeToConnectionStatus(): EventSubscription[] {
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/core/services/show-style-variant-state.service.ts]: No reason at the moment, removed!

File: src/app/core/services/show-style-variant-state.service.ts (hunk 16 above)
Comment (RasmusAlbrektsen) [reply src/app/core/services/show-style-variant-state.service.ts:32]: Actually yes, for now it is there to follow the structure of something having to happen on a reconnect event. I don't appreciate this boiler code though.

File: src/app/core/services/show-style-variant-state.service.ts [hunk 23]
```diff
@@ -0,0 +1,85 @@
...
+        return new BehaviorSubject<ShowStyleVariant>(showStyleVariant)
+    }
+
+    private async fetchShowStyleVariant(rundownId: string): Promise<ShowStyleVariant> {
+        return lastValueFrom(this.showStyleVariantService.getShowStyleVariant(rundownId))
+    }
+
+    private createUnsubscribeFromShowStyleVariantHandler(rundownId: string): () => void {
+        return () => this.unsubscribeFromShowStyleVariant(rundownId)
+    }
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/core/services/show-style-variant-state.service.ts]: I have changed it to no longer make use of the handler-method.

File: src/app/rundown-overview/components/rundown-overview/rundown-overview.component.html [hunk 24]
```diff
@@ -1,3 +1,5 @@
+<sofie-header></sofie-header>
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown-overview/components/rundown-overview/rundown-overview.component.html:1]: I can change it if you want? I just assumed it should be broad since it is the header for alle other pages than the rundown view.

File: src/app/core/services/show-style-variant-state.service.ts (hunk 16 above)
Comment (andr9528) [reply src/app/core/services/show-style-variant-state.service.ts:32]: Sound like we have the structure elsewhere, which means we can look it up, how it is suppose to be when needed. 
That also means that for now, you should be able to remove it without issue.

File: src/app/rundown/components/rundown-header/rundown-header.component.html (hunk 19 above)
Comment (KvelaGorrrrnio) [reply src/app/rundown/components/rundown-header/rundown-header.component.html:7]: `rundownPath` is a variable of `RundownHeaderComponent` that is passed to the input parameter `label` of `RundownHeaderPanelComponent`.

File: src/app/core/models/show-style-variant.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/app/core/models/show-style-variant.ts:20]: > Sidenote: We want to get rid of the Gfx and instead write Graphic. 

I have been enlightened

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+      this.setDesignFromInfinitePieces()
+      this.setSchemaFromInfinitePieces()
+    }
+  }
+
+  private setDefaultHeaderInformation(): void {
+    if (!this.showStyleVariant) {
+      return
+    }
+    const gfxDefaults = this.showStyleVariant.blueprintConfiguration.GfxDefaults[0]
```
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:81]: > Then we could also get rid of the Gfx abbreviation and name it GraphicDefaults instead.

I have been enlightened

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+
+  private isDesignInfinitePiece(piece: Piece): boolean {
+    return piece.name.startsWith('DESIGN_')
+  }
+
+  private isSkemaInfinitePiece(piece: Piece): boolean {
+    return piece.name.startsWith('DESIGN_')
+  }
+
+  private getGfxNameFromTemplate(template: string) {
```
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:112]: I have been enlightened

File: src/app/rundown/components/rundown-header/rundown-header.component.html [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+    <sofie-rundown-header-panel label="Design">{{design}}</sofie-rundown-header-panel>
+    <sofie-rundown-header-panel label="Local Time">{{currentLocalDate | date: 'H:mm:ss'}}</sofie-rundown-header-panel>
+
+    <a class="exit-button" title="Exit" routerLink="/rundowns">
+        <svg aria-hidden="true" width="1.500em" height="1.500em" viewBox="0 0 24 24">
+            <path fill="currentColor" fill-rule="evenodd" clip-rule="evenodd"
+                  d="M12 10.5858l6.2929-6.29289 1.4142 1.41421L13.4142 12l6.2929
+               6.2929-1.4142 1.4142L12 13.4142l-6.29288 6.2929-1.41421-1.4142L10.5858
+                12 4.29291 5.70712l1.41421-1.41421L12 10.5858z">
+            </path>
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.html]: This gave birth to `<icon-button>` which unfortunately requires two enums due to it using computed properties, but it serves the purpose quite well!

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+  public schema: string = ''
+  public design: string = ''
+
+  constructor(private showStyleVariantStateService: ShowStyleVariantStateService) {}
+
+  public ngOnInit(): void {
+    this.setRundownNameAndPath()
+    this.updateCurrentLocalDateIntervalId = setInterval(() =>
+      this.currentLocalDate = new Date()
+    , 1000)
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: Made into a constant instead!

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+    const infinitePieces: Piece[] = this.rundown.infinitePieces
+
+    const schemaPiece: Piece | undefined = infinitePieces.find((piece) => this.isSkemaInfinitePiece(piece))
+    if (schemaPiece) {
+      this.schema = schemaPiece.name
+    }
+  }
+
+  private isDesignInfinitePiece(piece: Piece): boolean {
+    return piece.name.startsWith('DESIGN_')
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: Moved into constant!

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: I made use of the suggestion!

File: src/app/rundown/components/rundown/rundown.component.html (hunk 14 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown/rundown.component.html]: I did exactly that, great suggestion!

File: src/app/core/services/show-style-variant-state.service.ts (hunk 16 above)
Comment (RasmusAlbrektsen) [reply src/app/core/services/show-style-variant-state.service.ts]: A very good point, i decided to remove most of the structure!

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts (hunk 17 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts]: Fixed!

File: src/app/shared/components/icon-button/icon-button.component.ts [hunk 30]
```diff
@@ -0,0 +1,57 @@
...
+    switch (iconButtonSize) {
+      case IconButtonSize.XS:
+        return "xs"
+      case IconButtonSize.SMALL:
+        return "sm"
+      case IconButtonSize.LARGE:
+        return "lg"
+      case IconButtonSize.XL:
+        return "xl"
+      case IconButtonSize.TWO_XL:
```
Comment (KvelaGorrrrnio) [inline src/app/shared/components/icon-button/icon-button.component.ts:50]: Is there a reason this is TWO_XL and not just `XXL`? :D

File: src/app/shared/components/icon-button/icon-button.component.ts [hunk 31]
```diff
@@ -0,0 +1,57 @@
...
+  public iconButton: IconButton
+
+  @Input()
+  public iconButtonSize?: IconButtonSize
+
+  public iconButtonProp: IconProp
+  public iconButtonSizeProp: SizeProp
+
+  constructor() {}
+
```
Comment (KvelaGorrrrnio) [inline src/app/shared/components/icon-button/icon-button.component.ts:23]: Can be removed.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts [hunk 32]
```diff
@@ -102,15 +108,19 @@ export class RundownHeaderComponent implements OnInit, OnDestroy, OnChanges {
...
 
   private getGfxNameFromTemplate(template: string) {
     const pattern: RegExp = /^.+_(?<gfxName>\w+)$/
-    return template.match(pattern)?.groups?.['gfxName'] ?? template
+    const { gfxName } = template.match(pattern)?.groups ?? {}
+    return gfxName ?? template
   }
+
+  protected readonly IconButton = IconButton;
+  protected readonly IconButtonSize = IconButtonSize;
```
Comment (LindvedKrvang) [inline src/app/rundown/components/rundown-header/rundown-header.component.ts:125]: Please place instance variables at the top of the class. This is for consistency between all our classes, so if we ever want to check the instance variables of a class, there is only one place for us to check.

Sidenote: What's the reason for making them protected?

File: src/app/rundown/components/rundown/rundown.component.html [hunk 33]
```diff
@@ -1,17 +1,13 @@
 <sofie-rundown-header *ngIf="rundown" class="c-rundown-header" [rundown]="rundown"></sofie-rundown-header>
 
 <div *ngIf="rundown?.isActive" class="test-buttons">
-    <ng-container *ngIf="!rundown?.isActive; else activeRundownButtons"></ng-container>
+    <button mat-raised-button color="primary" style="margin-left: 20px" (click)="takeNext()">Take Next</button>
```
Comment (LindvedKrvang) [inline src/app/rundown/components/rundown/rundown.component.html:4]: We should really begin to encapsulate components like this into our own components.
Just like you have done with the `IconButton`, we should also have a regular `Button` where the material design is encapsulated it.
Same goes for all the other components like `mat-card`, `mat-table` etc.

Not sure this is the PR to do it in, but it's an FYI

File: src/app/shared/components/icon-button/icon-button.component.scss [hunk 34]
```diff
@@ -0,0 +1 @@
+
```
Comment (LindvedKrvang) [inline src/app/shared/components/icon-button/icon-button.component.scss:1]: We should get in the happit of deleting empty SCSS files. We can always add them again if needed.

File: src/app/shared/components/icon-button/icon-button.component.ts [hunk 35]
```diff
@@ -0,0 +1,57 @@
...
+  ngOnInit(): void {
```
Comment (LindvedKrvang) [inline src/app/shared/components/icon-button/icon-button.component.ts:24]: Missing access level modifier

File: src/app/shared/components/icon-button/icon-button.component.ts (hunk 30 above)
Comment (LindvedKrvang) [reply src/app/shared/components/icon-button/icon-button.component.ts:50]: To piggyback on this. What is the reason for using both abbreviations and the fully written word i.e. you are using both `XS` and `SMALL`.
You should choose one and commit to it, so either you do `XS` and `S` or you do `EXTRA_SMALL` and `SMALL`

File: src/app/shared/enums/icon-button.ts [hunk 36]
```diff
@@ -0,0 +1,14 @@
+import {faTrashCan, faXmark} from "@fortawesome/free-solid-svg-icons";
```
Comment (LindvedKrvang) [inline src/app/shared/enums/icon-button.ts:1]: Unsused import. Can be deleted

**Review by LindvedKrvang — APPROVED** [review]
Looks good.
Now we just need to make more components like the `IconButton` :)

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 32 above)
Comment (KvelaGorrrrnio) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:125]: `protected` makes them available to the HTML template, but not outside components.

File: src/app/shared/components/icon-button/icon-button.component.ts (hunk 30 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/components/icon-button/icon-button.component.ts:50]: I believe i lost myself a bit in this. I for some reason made it fit the actual values that you pass to the size property, don't ask me why :D. I'll fix this in the next commit.

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts [hunk 37]
```diff
@@ -5,12 +5,11 @@ import {instance, mock, when} from "@typestrong/ts-mockito";
 import {RouterModule} from "@angular/router";
 import {RundownService} from "../../../core/abstractions/rundown.service";
 
-describe('HeaderContextMenuComponent', () => {
+describe('RundownHeaderContextMenuComponent', () => {
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts]: Changed!

File: src/app/shared/components/icon-button/icon-button.component.ts (hunk 30 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/components/icon-button/icon-button.component.ts]: I've changed it to be  `XS`, `S`, `M` and so forth, instead.

File: src/app/shared/components/icon-button/icon-button.component.ts (hunk 31 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/components/icon-button/icon-button.component.ts]: Removed!

File: src/app/shared/components/icon-button/icon-button.component.ts [hunk 38]
```diff
@@ -0,0 +1,57 @@
...
+      case IconButton.XMARK:
+        return faXmark
+      case IconButton.DELETE:
+        return faTrashCan
+      default:
+        return faXmark
+    }
+  }
+
+  public getIconSizeProperty(iconButtonSize?: IconButtonSize): SizeProp {
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/shared/components/icon-button/icon-button.component.ts:38]: This has been implemented!

File: src/app/shared/enums/icon-button.ts [hunk 39]
```diff
@@ -0,0 +1,14 @@
...
+    DELETE = 'delete',
+}
+
+export enum IconButtonSize {
+    XS = 'xs',
+    SMALL = 'sm',
+    LARGE = 'lg',
+    XL = 'xl',
+    TWO_XL = '2xl',
+}
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/shared/enums/icon-button.ts]: Changed so that `L = 'L'`!

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 32 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts:125]: Good catch, i missed these. They were auto generated at some point, and they are automatically put there at the bottom. I assume they are protected to limit exposure, but being private wouldn't let them be recognized by the AOT compiler.

File: src/app/rundown/components/rundown/rundown.component.html (hunk 33 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown/rundown.component.html:4]: Very good point, we should make a task for this, or write it into tasks that relates. I really think we should limit how much this PR is doing at the moment, it is part of the sprint goal and we should get this through first. 
I would gladly do this as part of something else, or as a standalone task at some point.

File: src/app/rundown/components/rundown-header/rundown-header.component.ts (hunk 32 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header/rundown-header.component.ts]: I have moved them now!

File: src/app/shared/components/icon-button/icon-button.component.scss (hunk 34 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/components/icon-button/icon-button.component.scss]: Good point, i deleted the file!

File: src/app/shared/components/icon-button/icon-button.component.ts (hunk 35 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/components/icon-button/icon-button.component.ts]: Inserted!

File: src/app/shared/enums/icon-button.ts (hunk 36 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/enums/icon-button.ts]: I have removed it, good catch!

**Review by KvelaGorrrrnio — APPROVED** [review]
